src/unix/solarish/mod.rs

```diff
@@ -3125,104 +3125,6 @@ extern "C" {
-    ) -> ::ssize_t;
-    // #include <link.h>
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-    pub fn dl_iterate_phdr(
-        callback: ::Option<
-            unsafe extern "C" fn(
-                info: *mut dl_phdr_info,
-                size: usize,
-                data: *mut ::c_void,
-            ) -> ::c_int,
-        >,
-        data: *mut ::c_void,
-    ) -> ::c_int;
-    pub fn getpagesize() -> ::c_int;
-    pub fn getpagesizes(pagesize: *mut ::size_t, nelem: ::c_int) -> ::c_int;
-    pub fn mmapobj(
-        fd: ::c_int,
-        flags: ::c_uint,
-        storage: *mut mmapobj_result_t,
-        elements: *mut ::c_uint,
-        arg: *mut ::c_void,
-    ) -> ::c_int;
-    pub fn meminfo(
-        inaddr: *const u64,
-        addr_count: ::c_int,
-        info_req: *const ::c_uint,
-        info_count: ::c_int,
-        outdata: *mut u64,
-        validity: *mut ::c_uint,
-    ) -> ::c_int;
-
-    pub fn strcasecmp_l(s1: *const ::c_char, s2: *const ::c_char, loc: ::locale_t) -> ::c_int;
-    pub fn strncasecmp_l(
-        s1: *const ::c_char,
-        s2: *const ::c_char,
-        n: ::size_t,
-        loc: ::locale_t,
-    ) -> ::c_int;
-    pub fn strsep(string: *mut *mut ::c_char, delim: *const ::c_char) -> *mut ::c_char;
-
-    pub fn getisax(array: *mut u32, n: ::c_uint) -> ::c_uint;
-
-    pub fn backtrace(buffer: *mut *mut ::c_void, size: ::c_int) -> ::c_int;
-    pub fn backtrace_symbols(buffer: *const *mut ::c_void, size: ::c_int) -> *mut *mut ::c_char;
-    pub fn backtrace_symbols_fd(buffer: *const *mut ::c_void, size: ::c_int, fd: ::c_int);
-}
-
-#[link(name = "lgrp")]
-extern "C" {
-    pub fn lgrp_init(view: lgrp_view_t) -> lgrp_cookie_t;
-    pub fn lgrp_fini(cookie: lgrp_cookie_t) -> ::c_int;
-    pub fn lgrp_affinity_get(
-        idtype: ::idtype_t,
-        id: ::id_t,
-        lgrp: ::lgrp_id_t,
-    ) -> ::lgrp_affinity_t;
-    pub fn lgrp_affinity_set(
-        idtype: ::idtype_t,
-        id: ::id_t,
-        lgrp: ::lgrp_id_t,
-        aff: lgrp_affinity_t,
-    ) -> ::lgrp_affinity_t;
-    pub fn lgrp_cpus(
-        cookie: ::lgrp_cookie_t,
-        lgrp: ::lgrp_id_t,
-        cpuids: *mut ::processorid_t,
-        count: ::c_uint,
-        content: ::lgrp_content_t,
-    ) -> ::c_int;
-    pub fn lgrp_mem_size(
-        cookie: ::lgrp_cookie_t,
-        lgrp: ::lgrp_id_t,
-        tpe: ::lgrp_mem_size_flag_t,
-        content: ::lgrp_content_t,
-    ) -> ::lgrp_mem_size_t;
-    pub fn lgrp_nlgrps(cookie: ::lgrp_cookie_t) -> ::c_int;
-    pub fn lgrp_view(cookie: ::lgrp_cookie_t) -> ::lgrp_view_t;
-    pub fn lgrp_home(idtype: ::idtype_t, id: ::id_t) -> ::lgrp_id_t;
-    pub fn lgrp_version(version: ::c_int) -> ::c_int;
-    pub fn lgrp_resources(
-        cookie: ::lgrp_cookie_t,
-        lgrp: ::lgrp_id_t,
-        lgrps: *mut ::lgrp_id_t,
-        count: ::c_uint,
-        tpe: ::lgrp_rsrc_t,
-    ) -> ::c_int;
-    pub fn lgrp_root(cookie: ::lgrp_cookie_t) -> ::lgrp_id_t;
-}
-
```

Comment:
sure I knew it would provoke reactions, easy to rollback as separate commit.

Reply:
I hope it get fixed in their CI side, I wish it s easy as it sounds, did not please me to remove these.

Reply:
> sure I knew it would provoke reactions, easy to rollback as separate commit.

Please tag us in the rust-lang/rust discussion next time.  We could more quickly address the problem directly.

This should not be committed as-is with plans for a rollback in a later commit.  Let's get the `::Copy` and '::Clone` fixes in (which are good, and needed), and then address the Solaris build issues separately.